Roles/Impostor/DollMaster.cs

```diff
@@ -0,0 +1,335 @@
+    }
+
+    private static void BootPossessedPlayerFromVentUpdate(PlayerControl target) // Boot Possessed Player from vent if inside of a vent and if waiting.
+    {
+        if (IsControllingPlayer && target.inVent && !target.walkingToVent)
+        {
+            _ = new LateTask(() =>
+            {
+                if (!target.inVent || target.walkingToVent) return;
+                target.MyPhysics.RpcBootFromVent(GetPlayerVentId(target));
+            }, 0.25f, "Boot Possessed Player from vent: " + GetPlayerVentId(target));
+        }
+    }
+
+    private static void WaitToUnPossessUpdate(PlayerControl pc, PlayerControl target) // If DollMaster can't be teleported start waiting to unpossess.
+    {
+        if (IsControllingPlayer && WaitToUnPossess)
+        {
+            // Boot DollMaster from vent if inside of a vent and if waiting.
+            if (pc.inVent && !pc.walkingToVent && !pc.MyPhysics.Animations.IsPlayingEnterVentAnimation())
+            {
+                _ = new LateTask(() =>
+                {
+                    if (!pc.inVent || pc.walkingToVent || pc.MyPhysics.Animations.IsPlayingEnterVentAnimation()) return;
+                    pc.MyPhysics.RpcBootFromVent(GetPlayerVentId(pc));
+                }, 0.3f, "Boot DollMaster from vent: " + GetPlayerVentId(pc));
+            }
+            // Unpossessed after waiting for DollMaster.
+            if (pc.CanBeTeleported())
+            {
+                _ = new LateTask(() =>
+                {
+                    if (!WaitToUnPossess) return;
+                    UnPossess(pc, target);
+                    GetPlayersPositions(pc);
+                    SwapPlayersPositions(pc);
+                }, 0.15f, "UnPossess");
+            }
+        }
+    }
+
+    // Get vent Id that the player is in.
+    private static int GetPlayerVentId(PlayerControl pc)
+    {
+        if (!(ShipStatus.Instance.Systems.TryGetValue(SystemTypes.Ventilation, out var systemType) &&
+              systemType.TryCast<VentilationSystem>() is VentilationSystem ventilationSystem))
+            return 0;
+
+        return ventilationSystem.PlayersInsideVents.TryGetValue(pc.PlayerId, out var playerIdVentId) ? playerIdVentId : 0;
+    }
+
+    // Prepare for a meeting if possessing.
+    public override void OnReportDeadBody(PlayerControl pc, PlayerControl target) // Fix crap when meeting gets called.
+    {
+        if (IsControllingPlayer && controllingTarget != null && DollMasterTarget != null)
+        {
+            bool shouldAnimate = false;
+            Utils.GetPlayerById(DollMasterTarget.PlayerId).RpcShapeshift(Utils.GetPlayerById(DollMasterTarget.PlayerId), shouldAnimate);
+            Utils.GetPlayerById(controllingTarget.PlayerId).RpcShapeshift(Utils.GetPlayerById(controllingTarget.PlayerId), shouldAnimate);
+            UnPossess(Utils.GetPlayerById(DollMasterTarget.PlayerId), Utils.GetPlayerById(controllingTarget.PlayerId));

```

Comment:
I don't think you need to use `GetPlayerById` because they already contain `PlayerControl`

![image](https://github.com/0xDrMoe/TownofHost-Enhanced/assets/104814436/2593e98e-cf99-4d1d-8b2b-cf145d0e1307)

Reply:
There are similar cases in many parts of the code

Reply:
Yeah sorry about that this is one of my first roles at the time and I was using a different method to get the player IDs back then, so I'm going to do a general cleanup right now.